### PR TITLE
Migrate to Kotlin 1.4.21, drop 1.3.72 version, improve build-scripts, update Gradle, fix broken tests

### PR DESCRIPTION
**Pull request by CommanderTvis**

### CODE REVIEW

**Comment by CommanderTvis** [comment]
@h0tk3y any updates?

**Comment by rnett** [comment]
@CommanderTvis have you thought about publishing your branch somewhere eventually?

**Comment by CommanderTvis** [comment]
Currently, no, since setting up my repository will add new changes to this PR

File: README.md [hunk 1]
```diff
@@ -28,13 +28,11 @@ val ast = booleanGrammar.parseToEnd("a & !b | b & (!a | c)")
...
 ### Using with Gradle
 
-> Use version 0.4.0 with Kotiln 1.3.72 and version 0.4.0-1.4-M2 with Kotlin 1.4-M2.
-
 Add the JCenter repository:
 
 ```groovy
 repositories {
-    maven { setUrl("https://dl.bintray.com/hotkeytlt/maven") }
+    maven { url 'https://dl.bintray.com/hotkeytlt/maven' }
```
Comment (h0tk3y) [inline README.md:35]: I used this option, as this is the only syntax that I know that works well with both Groovy and KTS build scripts. I'd suggest that this line stays as is.

File: buildSrc/src/main/kotlin/MultiplatformLibraryDependency.kt [hunk 2]
```diff
@@ -1,47 +1,30 @@
 import org.gradle.api.Project
 
-val Project.kotlinVersion: String
+inline val Project.kotlinVersion: String
```
Comment (h0tk3y) [inline buildSrc/src/main/kotlin/MultiplatformLibraryDependency.kt:3]: Why do these vals need to be inline?

File: README.md (hunk 1 above)
Comment (CommanderTvis) [reply README.md:35]: OK

**Comment by h0tk3y** [comment]
Thanks a lot @CommanderTvis! I've manually picked a lot of your changes that I found useful. I had a different view on some of them, though, that's why I decided to do that manually. I've also cherry-picked one of your commits to ensure that you get to the repository contributors.

**Comment by h0tk3y** [comment]
I'll publish the new version to Maven Central soon, maybe today.

**Pull request closed without merging**